csharp/src/Google.Protobuf/JsonFormatter.cs

```diff
@@ -795,10 +807,23 @@ public Settings(bool formatDefaultValues) : this(formatDefaultValues, TypeRegist
             /// </summary>
             /// <param name="formatDefaultValues"><c>true</c> if default values (0, empty strings etc) should be formatted; <c>false</c> otherwise.</param>
             /// <param name="typeRegistry">The <see cref="TypeRegistry"/> to use when formatting <see cref="Any"/> messages.</param>
-            public Settings(bool formatDefaultValues, TypeRegistry typeRegistry)
+            public Settings(bool formatDefaultValues, TypeRegistry typeRegistry) : this(formatDefaultValues, typeRegistry, false)
+            {
+            }
+
+            /// <summary>
+            /// Creates a new <see cref="Settings"/> object with the specified parameters.
+            /// </summary>
+            /// <param name="formatDefaultValues"><c>true</c> if default values (0, empty strings etc) should be formatted; <c>false</c> otherwise.</param>
+            /// <param name="typeRegistry">The <see cref="TypeRegistry"/> to use when formatting <see cref="Any"/> messages. TypeRegistry.Empty will be used if it is null.</param>
+            /// <param name="formatEnumsAsIntegers"><c>true</c> format the enums as ints; <c>false</c> otherwise.</param>
+            public Settings(bool formatDefaultValues = false,
```

Comment:
I suggest you do it in this PR - I can't add a `WithFormatEnumsAsIntegers` until there's such a constructor parameter, for example. I could do it for the other two, but it's really simple:

```csharp
// TODO: Documentation
public Settings WithFormatDefaultValues(bool formatDefaultValues) =>
    new Settings(formatDefaultValues, TypeRegistry);

public Settings WithTypeRegistry(TypeRegistry typeRegistry) =>
    new Settings(FormatDefaultValues, typeRegistry);
```